sys/kern/vfs.c

```diff
@@ -343,24 +344,16 @@ int vfs_namelookup(const char *path, vnode_t **vp) {
   return error;
 }
 
-int vfs_namecreate(const char *path, vnode_t **dvp, componentname_t *cn) {
+int vfs_namecreate(const char *path, vnode_t **dvp, vnode_t **vp,
```

Comment:
I wonder... why signature of this function doesn't look like this: `int vfs_namecreate(const char *path, vnode_t **dvp, vnode_t **vp, const char **namep)`. Won't `cn` always be a copy of last component from `path`? Why are we even consider copying it? Such change would simplify a call to `VOP_CREATE` as you would not need to create `namecopy`.

Reply:
During recent face-to-face discussion we came to a conclusion that some `vnodeops` must be modified to accept `componentname_t *` instead of `const char *` to make this code not to use dynamic allocation. Please address that in near future.